great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -849,17 +851,11 @@ def get_datasource(self, name: str = "default") -> FluentDatasource:
         Raises:
             ValueError: The input `datasource_name` is None.
         """
-        if name is None:
-            raise ValueError("Must provide a datasource_name to retrieve an existing Datasource")  # noqa: TRY003
-
         try:
-            datasource = self.data_sources.all()[name]
+            return self.data_sources.get(name)
```

Comment:
Could we emit a deprecation warning here?

Reply:
The decorator only updates how it presents in docs iirc

Reply:
Oh good call. I think the docstring for the decorator says it updates the docstring, but I guess that's only in the docs. Added a warning on there. Thanks!

Reply:
Could we add a `# deprecated-v1.2.0` comment above this to be consistent with other warnings?